server/src/test/java/keywhiz/service/resources/admin/SecretsResourceIntegrationTest.java

```diff
@@ -43,16 +47,49 @@
     keywhizClient = TestClients.keywhizClient();
   }
 
+  @Test
+  public void createsSecretWithOwner() throws Exception {
+    login();
+
+    String owner = UUID.randomUUID().toString();
+
+    keywhizClient.createGroup(owner, "description", ImmutableMap.of());
+
+    SecretDetailResponse response = keywhizClient.createSecret(
+        "name",
+        owner,
+        "description",
+        "content".getBytes(StandardCharsets.UTF_8),
+        ImmutableMap.of(),
+        0);
+    assertEquals(owner, response.owner);
+  }
```

Comment:
This is the test I should have started with: an end-to-end integration test that operates at the very highest level (here, the client API level). With this in place I'm much more confident that there isn't some other propagation bug lurking in the codebase.